assemblyline/filestore/__init__.py

```diff
@@ -140,7 +140,7 @@ def create_transport(url, connection_attempts=None):
         valid_bool_keys = ['allow_directory_access']
         extras = _get_extras(parse_qs(parsed.query), valid_str_keys=valid_str_keys, valid_bool_keys=valid_bool_keys)
 
-        t = TransportAzure(base=base, host=host, connection_attempts=connection_attempts, **extras)
+        t = TransportAzure(base=base, host=host, connection_attempts=connection_attempts, use_mi=use_mi, **extras)
```

Comment:
Here instead of getting use_mi from the create_transport function, you can simply add use_mi to the valid_bool_keys. This will allow you to simply add this new boolean to the filestore url. Ex:
azure://my.storage.url/storage/?use_mi=true